packages/react/dismissable-layer/src/DismissableLayer.tsx

```diff
@@ -155,6 +161,40 @@ const DismissableLayer = React.forwardRef<DismissableLayerElement, DismissableLa
 
 DismissableLayer.displayName = DISMISSABLE_LAYER_NAME;
 
+/* -------------------------------------------------------------------------------------------------
+ * DismissableLayerBranch
+ * -----------------------------------------------------------------------------------------------*/
+
+const BRANCH_NAME = 'DismissableLayerBranch';
+
+type DismissableLayerBranchElement = React.ElementRef<typeof Primitive.div>;
+interface DismissableLayerBranchProps extends PrimitiveDivProps {}
+
+const DismissableLayerBranch = React.forwardRef<
+  DismissableLayerBranchElement,
+  DismissableLayerBranchProps
+>((props, forwardedRef) => {
+  const context = React.useContext(DismissableLayerContext);
+  const ref = React.useRef<DismissableLayerBranchElement>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
+
+  React.useEffect(() => {
+    const node = ref.current;
+    if (node) {
+      context.branches.add(node);
+      dispatchUpdate();
```

Comment:
Why do we need to trigger an update here in this case?
This doesn't affect the layering does it?

Reply:
If a `Branch` is rendered after a `DismissableLayer` it should be considered inside all of the dismissable layers currently rendered. I did this to account for generic use-cases, not just `Toast`.